internal/moduletest/graph/apply.go

```diff
@@ -115,6 +119,34 @@ func (n *NodeTestRun) testApply(ctx *EvalContext, variables terraform.InputValue
 	// actually updated by this change. We want to use the run that
 	// most recently updated the tracked state as the cleanup
 	// configuration.
+	if state := ctx.GetFileState(key); state.backend.instance != nil {
+		if state.backend.run.Name == run.Name {
+			// We've just done an apply for the run that contains the backend block.
+			// This run block is allowed to update the remote state if any changes have occurred, e.g:
+			//   1) Test is being run for the first time, and state is being created in the backend
+			//   2) Config changes mean that long-lived resources need to be updated, and state updated too.
+			//   3) Drift may have occurred, and this step of the test corrects remote objects to resemble config.
+			b := state.backend.instance
+			stateMgr, err := b.StateMgr(backend.DefaultStateName) // We only allow use of the default workspace
```

Comment:
Should we add code to detect if the user has another workspace selected and warn them that test uses a default workspace?